src/tasks/MotionForceTask.h

```diff
@@ -664,6 +654,69 @@ class MotionForceTask : public TemplateTask {
 		return _partial_task_projection.block<3, 3>(3, 3);
 	}
 
+	// -------- singularity handling methods --------
+
+	/**
+	 * @brief 	Set the Dynamic Decoupling Type. See the definition of the
+	 * DynamicDecouplingType enum for more details
+	 *
+	 * @param type Dynamic decoupling type 
+	 */
+	void setDynamicDecouplingType(const DynamicDecouplingType type) {
+		_singularity_handler->setDynamicDecouplingType(type);
+	}
+
+    /**
+     * @brief Enforces type 1 handling behavior if set to true, otherwise handle 
+     * type 1 or type 2 as usual
+     * 
+     * @param flag true to enforce type 1 handling behavior 
+     */
+	void handleAllSingularitiesAsType1(const bool flag) {
+		_singularity_handler->handleAllSingularitiesAsType1(flag);
+	}
+	
+	/**
+	 * @brief Set the desired posture for type 1 singularity handling  
+	 * 
+	 * @param q_des desired posture 
+	 */
+	void setType1Posture(const VectorXd& q_des) {
+		_singularity_handler->setType1Posture(q_des);
+	}
+
+	/**
+	 * @brief Enables or disables singularity handling in singularity handler  
+	 * 
+	 * @param flag true to enable singularity handling, if false the behavior is just task truncation 
+	 */
+	void enableSingularityHandling(const bool flag) {
```

Comment:
make 2 functions
enableSingularityHandling()
disableSingularityHandling()
that take no argument. I think it is cleaner and mimics the velocity saturation and otg enable/disable